lib/runtime/helpers_unchecked_extrinsic_test.go

```diff
@@ -0,0 +1,228 @@
+	require.Equal(t, []byte{0, 1, 1, 0, 5}, res)
+
+	// these aren't working, probably due to above issue
+	//val, err := rt.storage.GetStorage([]byte("testkey"))
+	//require.NoError(t, err)
+	//require.Equal(t, []byte("testvalue"), val)
+	//
+	//for i := 0; i < maxRetries; i++ {
+	//	_, err = rt.FinalizeBlock()
+	//	if err == nil {
+	//		break
+	//	}
+	//}
+	//require.NoError(t, err)
+	//
+	//val, err = rt.storage.GetStorage([]byte("testkey"))
+	//require.NoError(t, err)
+	//// TODO: why does calling finalize_block modify the storage?
+	//require.NotEqual(t, []byte("testvalue"), val)
+}
+
+func TestApplyExtrinsic_Transfer_NoBalance_UncheckedExt(t *testing.T) {
+	rt := NewTestRuntime(t, NODE_RUNTIME)
+
+	// genesisHash from runtime must match genesisHash used in transfer payload message signing
+	// genesis bytes for runtime seems to be stored in allocated storage at key 0xdfa1667c116b77971ada377f9bd9c485a0566b8e477ae01969120423f2f124ea
+	//  runtime trace logs show genesis set to below:
+	genesisBytes := []byte{83, 121, 115, 116, 101, 109, 32, 66, 108, 111, 99, 107, 72, 97, 115, 104, 0, 0, 0, 0, 101, 117, 101, 50, 54, 188, 7, 185, 79, 198, 211, 234}
+	genesisHash := common.BytesToHash(genesisBytes)
+
+	// Init transfer
+	header := &types.Header{
+		ParentHash: genesisHash,
+		Number:     big.NewInt(1),
+	}
+	err := rt.InitializeBlock(header)
+	require.NoError(t, err)
+
+	bob := kr.Bob.Public().Encode()
+	bb := [32]byte{}
+	copy(bb[:], bob)
+
+	var nonce uint64 = 0
+	tranCallData := struct {
+		Type byte
+		To   [32]byte
+		Amt  *big.Int
+	}{
+		Type: byte(255), // TODO determine why this is 255 (Lookup type?)
+		To:   bb,
+		Amt:  big.NewInt(1234),
+	}
+	transferF := &extrinsic.Function{
+		Call:     extrinsic.Balances,
+		Pallet:   extrinsic.PB_Transfer,
+		CallData: tranCallData,
+	}
+
+	ux, err := extrinsic.CreateUncheckedExtrinsic(transferF, new(big.Int).SetUint64(nonce), genesisHash, kr.Alice)
+	require.NoError(t, err)
+
+	uxEnc, err := ux.Encode()
+	require.NoError(t, err)
+
+	res, err := rt.ApplyExtrinsic(uxEnc)
+	require.NoError(t, err)
+
+	require.Equal(t, []byte{1, 2, 0, 1}, res) // 0x01020001 represents Apply error, Type: Payment: Inability to pay (expected result)
+}
+
+func TestApplyExtrinsic_Transfer_WithBalance_UncheckedExtrinsic(t *testing.T) {
+	rt := NewTestRuntime(t, NODE_RUNTIME)
+
+	genesisBytes := []byte{83, 121, 115, 116, 101, 109, 32, 66, 108, 111, 99, 107, 72, 97, 115, 104, 0, 0, 0, 0, 101, 117, 101, 50, 54, 188, 7, 185, 79, 198, 211, 234}
+	genesisHash := common.BytesToHash(genesisBytes)
+
+	// Init transfer
+	header := &types.Header{
+		ParentHash: genesisHash,
+		Number:     big.NewInt(1),
+	}
+	err := rt.InitializeBlock(header)
+	require.NoError(t, err)
+
+	alice := kr.Alice.Public().Encode()
+	bob := kr.Bob.Public().Encode()
+
+	ab := [32]byte{}
+	copy(ab[:], alice)
+
+	bb := [32]byte{}
+	copy(bb[:], bob)
+
+	rt.storage.SetBalance(ab, 2000)
```

Comment:
I think the balance key is different with the new runtime. you might want to try updating it by looking at the runtime logs to see what key it gets before failing with no balance. however might still be blocked by runtime issue

Reply:
Right, I think so, the trace logs have gold in them.  I'll dig into them for future PR with balance transfer with balance.